src/NHibernate/Criterion/InExpression.cs

```diff
@@ -64,24 +64,22 @@ public override SqlString ToSqlString(ICriteria criteria, ICriteriaQuery criteri
 
 			//TODO: add default capacity
 			SqlStringBuilder result = new SqlStringBuilder();
-			SqlString[] columnNames =
-				CriterionUtil.GetColumnNames(_propertyName, _projection, criteriaQuery, criteria);
+			var columnNames =
+				CriterionUtil.GetColumnNamesAsSqlStringParts(_propertyName, _projection, criteriaQuery, criteria);
```

Comment:
Changes in this file conflict with #2158. To me it seems those changes should be dropped. I let you see by yourself.

Reply:
Done. Yeah just dropped them